src/features/Cart/hooks/useCart.ts

```diff
@@ -0,0 +1,87 @@
+import { useContext, useEffect, useRef, useState } from 'react';
+
+import { deleteCartItem, getCartItemList, updateCartItem } from '@/api/cart';
+import { ToastContext } from '@/shared/context/ToastProvider';
+import { isError } from '@/shared/utils/isError';
+import { useFetchData } from '@/shared/hooks/useFetchData';
+import { CartItem } from '../types/Cart.types';
+
+export const useCart = () => {
+  const cart = useFetchData<CartItem[]>({ autoFetch: getCartItemList });
+  const { showToast } = useContext(ToastContext);
+  const [checkedItems, setCheckedItems] = useState<Set<number>>(new Set());
+  const hasInitialized = useRef(false);
+
+  useEffect(() => {
+    if (cart.data && cart.data.length > 0 && !hasInitialized.current) {
+      setCheckedItems(new Set(cart.data.map((item) => item.id)));
+      hasInitialized.current = true;
+    }
+  }, [cart.data, checkedItems.size]);
```

Comment:
checkdItems를 상태가 아니라 값으로 관리하면 어떨까요?

```suggestion
  const checkedItems = useMemo(() => new Set(cart.data.map((item) => item.id)), [cart.data])
```

이러면 useEffect를 사용할 필요가 없답니다 ㅎㅎ

Reply:
현재 체크 상태는 사용자 상호작용(개별/전체 토글 등)에 따라 동적으로 변경되기 때문에,
상태로 관리해야 유연하게 처리할 수 있다고 판단했습니다!

말씀해주신 `useMemo` 방식은 `cart.data` 변화에만 반응하기 때문에 사용자 조작이 반영되기는 힘들 것 같은데 맞을까요 ??